### DISCUSSION TITLE
Dynamically update registry metadata?

### DISCUSSION THREAD

**Discussion by bhoriuchi** [micro/go-micro#2266] [Q&A]
Quick question. I would like to update metadata dynamically based on certain conditions on a micro service server to use in concert with some custom registry selection.

I do not currently see a way other than de-registering and re-registering the service.

Is that the approach I should take or is there something I am overlooking as far as updating metadata without re-registering.

For context I am using consul for registry

-- update --

after some testing it looks like metadata can be updated by simply updating the metadata options value and the register operation is  periodic and configurable by default. would be nice if the interface allowed you to immediately update the value.

**Comment by polanskya** [comment]
@bhoriuchi .. Do you have a code sample how you managed  top update registry service meta?

I tried to do the following , bit no luck.. I am using etcd registry.. 

`
s, _ := g.core.Registry().ListServices()
	 for _, v := range s {
		 if v.Name == "my.service" && v.Metadata == nil {
			v.Metadata = make(map[string]string)
			v.Metadata["log_level"] = "info"
			err := g.core.Registry().Register(v)
			
}
`
